sources/lib/reasoners/intervalCalculus.ml

```diff
@@ -1423,6 +1429,55 @@ let rec loop_update_intervals are_eq env cpt =
   then env
   else loop_update_intervals are_eq env cpt
 
+let calc_pow a b ty uf =
+  let ra, expl_a = Uf.find uf a in
+  let rb, expl_b = Uf.find uf b in
+  let pa = poly_of ra in
+  let pb = poly_of rb in
+  try
+    match P.is_const pb with
+    | Some c_y ->
+      let res,expl =
+        (* x¹ -> x *)
+        if Q.equal c_y Q.one then
+          ra, Ex.union expl_a expl_b
+          (* x⁰ -> 1 *)
+        else if Q.equal c_y Q.zero then
+          alien_of (P.create [] Q.one ty), Ex.union expl_a expl_b
+        else
+          match P.is_const pa with
+          | Some c_x ->
+            (* x^y *)
+            let res = Arith.calc_power c_x c_y ty  in
+            alien_of (P.create [] res ty), Ex.union expl_a expl_b
+          | None -> raise Exit
+      in
+      Some (res,expl)
```

Comment:
`Ex.union expl_a expl_b` can be compute here before returning `Some (res, ...)`: it's always the same in different cases